packages/integration-karma/test/shadow-dom/Event-properties/Event.target.spec.js

```diff
@@ -1,97 +1,97 @@
-    triggerElm.click();
-    expect(elm.syncTargetIsCorrect).toBe(true);
-    return new Promise(setTimeout).then(() => {
-        expect(elm.asyncTargetIsCorrect).toBe(true);
-    });
-});
 
-it('parent should receive composed event with correct target', function () {
-    const elm = createElement('x-parent', { is: XEventHandlingParent });
-    document.body.appendChild(elm);
-    const child = elm.shadowRoot.querySelector('x-event-dispatching-child');
-    child.dispatchFoo();
-    expect(elm.evtTargetIsXChild).toBe(true);
-    return Promise.resolve().then(() => {
-        expect(elm.shadowRoot.querySelector('.evt-target-is-x-child')).not.toBe(null);
+import Container from 'x/container';
+
+it('should retarget', (done) => {
+    const container = createElement('x-container', { is: Container });
+    document.body.appendChild(container);
+
+    const child = container.shadowRoot.querySelector('x-child');
+    child.addEventListener('test', (event) => {
+        expect(event.target).toEqual(child);
+        done();
     });
+
+    const div = child.shadowRoot.querySelector('div');
+    div.dispatchEvent(new CustomEvent('test', { bubbles: true, composed: true }));
 });
 
-describe('event.target on document event listener', () => {
-    let actual;
-    const listener = (evt) => {
-        actual = evt.target.tagName.toLowerCase();
-    };
-    beforeAll(() => {
-        document.addEventListener('click', listener);
-    });
-    afterAll(() => {
-        document.removeEventListener('click', listener);
-    });
-    it('should return correct target', function () {
-        const elm = createElement('x-document-event-listener', { is: XDocumentEventListener });
-        document.body.appendChild(elm);
-        elm.shadowRoot.querySelector('button').click();
-        expect(actual).toBe('x-document-event-listener');
+it('should retarget to the root component when accessed asynchronously', () => {
+    const container = createElement('x-container', { is: Container });
+    document.body.appendChild(container);
+
+    let event;
+    const child = container.shadowRoot.querySelector('x-child');
+    child.addEventListener('test', (e) => {
+        event = e;
     });
+
+    const div = child.shadowRoot.querySelector('div');
+    div.dispatchEvent(new CustomEvent('test', { bubbles: true, composed: true }));
+
+    expect(event.target).toEqual(container);
 });
 
-// legacy usecases
-describe('should not retarget event', () => {
-    if (!process.env.NATIVE_SHADOW) {
-        let elm;
-        let child;
-        let originalTarget;
+it('should retarget when accessed in a document event listener', (done) => {
+    const container = createElement('x-container', { is: Container });
+    document.body.appendChild(container);
+
+    document.addEventListener(
+        'test',
+        (event) => {
+            expect(event.target).toEqual(container);
+            done();
+        },
+        { once: true }
```

Comment:
🎉

Reply:
Turns out this option doesn't work in IE11